src/main/java/io/cryostat/rules/Rule.java

```diff
@@ -219,4 +230,117 @@ public String toString() {
             return getSerialKey();
         }
     }
+
+    public static Rule buildRule(MultiMap formAttributes) {
+        Rule.Builder builder =
+                new Rule.Builder()
+                        .name(formAttributes.get(Rule.Attribute.NAME.getSerialKey()))
+                        .targetAlias(formAttributes.get(Rule.Attribute.TARGET_ALIAS.getSerialKey()))
+                        .description(formAttributes.get(Rule.Attribute.DESCRIPTION.getSerialKey()))
+                        .eventSpecifier(
+                                formAttributes.get(Rule.Attribute.EVENT_SPECIFIER.getSerialKey()));
+
+        builder = setOptionalInt(builder, Rule.Attribute.ARCHIVAL_PERIOD_SECONDS, formAttributes);
+        builder = setOptionalInt(builder, Rule.Attribute.PRESERVED_ARCHIVES, formAttributes);
+        builder = setOptionalInt(builder, Rule.Attribute.MAX_AGE_SECONDS, formAttributes);
+        builder = setOptionalInt(builder, Rule.Attribute.MAX_SIZE_BYTES, formAttributes);
+
+        return builder.build();
+    }
+
+    public static Rule buildRule(JsonObject jsonObj) throws IllegalArgumentException {
+
+        Rule.Builder builder =
+                new Rule.Builder()
+                        .name(jsonObj.get(Rule.Attribute.NAME.getSerialKey()).getAsString())
+                        .targetAlias(
+                                jsonObj.get(Rule.Attribute.TARGET_ALIAS.getSerialKey())
+                                        .getAsString())
+                        .description(
+                                jsonObj.get(Rule.Attribute.DESCRIPTION.getSerialKey())
+                                        .getAsString())
+                        .eventSpecifier(
+                                jsonObj.get(Rule.Attribute.EVENT_SPECIFIER.getSerialKey())
+                                        .getAsString());
+        builder = setOptionalInt(builder, Rule.Attribute.ARCHIVAL_PERIOD_SECONDS, jsonObj);
+        builder = setOptionalInt(builder, Rule.Attribute.PRESERVED_ARCHIVES, jsonObj);
+        builder = setOptionalInt(builder, Rule.Attribute.MAX_AGE_SECONDS, jsonObj);
+        builder = setOptionalInt(builder, Rule.Attribute.MAX_SIZE_BYTES, jsonObj);
+
+        return builder.build();
+    }
+
+    public static Rule.Builder setOptionalInt(
```

Comment:
This part I'm a little more curious about refactoring actually, now that I see it. It takes a `Rule.Builder` as its first parameter and then returns that same `Rule.Builder` instance at the end so that it can be used as a fluent API, but it also lives within the `Rule` class. Maybe this should be moved inside the `Rule.Builder` and made non-static, and instead of taking a `Rule.Builder` reference as the first parameter, it can just operate on `this`?

Reply:
Would it look similar to this format?
```
public static class Builder {
  ...
 private setOptionalInt(Rule.Attribute key, MultiMap formAttributes)
            throws IllegalArgumentException {
   ...
   this = fn.apply(value);
}
...
}
```

Reply:
You can't assign to `this`, but I think what you intend is that the last line of that method would just be `return fn.apply(value)`. The `fn` comes from your `selectAttribute` method, which returns a Function/method reference to setter-methods from the Builder itself, which all return a reference to their Builder instance.

`selectAttribute` could also be moved into the `Builder` class, by the way.